test/lib/cli/commands/build.js

```diff
@@ -30,19 +30,19 @@ const defaultBuilderArgs = {
 	excludedTasks: undefined
 };
 
-test.before("Mocking before test execution", (t) => {
+test.beforeEach("Mocking before test execution", (t) => {
 	normalizerStub = sinon.stub(normalizer, "generateProjectTree");
 	builderStub = sinon.stub(builder, "build").returns(Promise.resolve());
 	loggerStub = sinon.stub(logger, "setShowProgress");
 });
 
-test.after("Restore mocks after test exection", (t) => {
+test.afterEach("Restore mocks after test exection", (t) => {
 	loggerStub.restore();
```

Comment:
We could also use `sinon.restore()` in a [`test.afterEach.always`](https://github.com/avajs/ava/tree/v0.25.0#before--after-hooks) callback as described here:
https://sinonjs.org/releases/v7.2.2/general-setup/

But we can also change that in a follow up PR.

Reply:
Great, I like it 👍 
Fixed in b282427

Reply:
I think it would be even better to use `test.afterEach`**`.always`**. That way, even if a test fails, the stubs created in that test are reset and can't influence any other tests.

Also, I'm not sure whether you need to supply a message to the beforeEach and afterEach hooks. I couldn't find this mentioned anywhere in the ava docs.